proposals/store_kyverno_policies_in_oci_registries.md

```diff
@@ -0,0 +1,129 @@
+
+    “The OCI Technical Oversight Board (TOB) has approved a new Artifacts project, utilizing the OCI manifest and OCI index definitions, new artifact types can be stored and served using the OCI distribution-spec without changing the actual distribution spec. This repository will provide a reference for artifact authors and registry implementers for supporting these new artifact types themselves with the existing implementations of distribution.”
+
+The most helpful advantage of having an OCI Artifacts definition is that the community can continue to innovate, focusing on new artifact types without building another storage solution (YASS). More technically, OCI Artifacts are not a new specification, format, or API, but rather a set of conventions for storing assets other than images inside an OCI registry. For example, thanks to OCI Artifacts spec, additional artifacts, such as Helm Charts, can be easily stored and accessible from an OCI registry.
+
+# Definitions
+
+[definitions]: #definitions
+
+Make a list of the definitions that may be useful for those reviewing. Include phrases and words that Kyverno users or other interested parties may not be familiar with.
+
+# Motivation
+
+[motivation]: #motivation
+
+OCI Artifacts provides a [reference guide](https://github.com/opencontainers/artifacts/blob/main/artifact-authors.md) for authors looking to introduce a new artifact type. A unique artifact type is similar to defining a file extension. Defining a unique artifact allows various tools to work with the type uniquely. For example, it will enable a registry to display the type and tooling, such as vulnerability scanners, to know if and how they should interact with the contents. To learn more about defining your artifact type, please [see](https://github.com/opencontainers/artifacts/blob/main/artifact-authors.md#defining-a-unique-artifact-type).
+
+     “application/vnd.[org/company].[objectType].[optional sub type].config.[version]+json”
+
+# Proposal
+
+> Issue [#3154](https://github.com/kyverno/kyverno/issues/3154)
+
+Two implementation points exist in the Kyverno space that can be used to support pulling/pushing policies based on OCI Artifacts specification: Kyverno itself and its Command Line Interface (CL)I. First, we will start by introducing new subcommands for pulling/pushing policies using Kyverno CLI and then to Kyverno itself.
+
+One of the initial steps that OCI artifact authors need to decide upon is the mediaTypes that we will use to store Kyverno policies. According to the OCI Artifact Authors specification format defined for new artifact types, for the Kyverno policy layer, which we use to keep the policy itself, it should be represented as application/vnd.cncf.kyverno.policy.layer.v1+yaml, and application/vnd.cncf.kyverno.config.v1+json which will be used to store the policy config.
```

Comment:
I've also added new mediaTypes for addressing a bunch of other use cases, here is the complete list:

```shell
# for policies in YAML format
application/vnd.cncf.kyverno.policy.layer.v1+yaml

# for policies in JSON format
application/vnd.cncf.kyverno.policy.layer.v1+json

# for storing multiple policies in the same layer
application/vnd.cncf.kyverno.policy.layer.v1.tar+gzip 
```

Reply:
@developer-guy - its not clear why we need new mediaTypes for Kyverno. 

Is there an existing media type that can be cleanly and securely reused?

Reply:
> Is there an existing media type that can be cleanly and securely reused?

Of course, we can use the existing mediaTypes defined in the OCI [image-spec](https://github.com/opencontainers/image-spec/blob/main/media-types.md).

> its not clear why we need new mediaTypes for Kyverno.

The intent was behind defining unique mediaTypes well defined in the [OCI Artifacts repository](https://github.com/opencontainers/artifacts/blob/main/artifact-authors.md#defining-a-unique-artifact-type). 

TLDR; 
Defining a unique artifact allows various tools to know how to uniquely work with the type.  It allows a registry to display the type and tooling, such as vulnerability scanners, a means to know if and how they should interact with the contents.

But we can continue with existing mediaTypes if this is the case, that would not be a problem. @JimBugwadia